src/index/hnsw.cpp

```diff
@@ -238,10 +238,14 @@ HNSW::knn_search(const DatasetPtr& query,
         auto params = HnswSearchParameters::FromJson(parameters);
 
         // perform search
+        int64_t original_k = k;
         std::priority_queue<std::pair<float, size_t>> results;
         double time_cost;
         try {
             Timer t(time_cost);
+            if (use_conjugate_graph_ and params.use_conjugate_graph_search) {
+                k = LOOK_AT_K;
+            }
```

Comment:
here adjust k to provide enough information for conjugate graph to enhance result